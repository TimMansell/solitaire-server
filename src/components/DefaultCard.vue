<template>
  <SvgIcon :name="value" />
</template>

<script>
import SvgIcon from '@/components/SvgIcon.vue';

export default {
  name: 'Card',
  components: {
    SvgIcon,
  },
  props: {
    value: {
      type: String,
      default: 'Aâ™£',
    },
  },
};
</script>
