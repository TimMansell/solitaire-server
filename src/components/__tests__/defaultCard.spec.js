import { shallowMount } from '@vue/test-utils';
import DefaultCard from '@/components/DefaultCard.vue';

describe('DefaultCard.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(DefaultCard);

    expect(wrapper).toMatchSnapshot();
  });

  it('should have correct props', () => {
    const propsData = {
      value: 'J',
      suit: 'â™£',
    };

    const wrapper = shallowMount(DefaultCard, {
      propsData,
    });

    expect(wrapper.props().value).toBe(propsData.value);
    expect(wrapper.props().suit).toBe(propsData.suit);
  });
});
