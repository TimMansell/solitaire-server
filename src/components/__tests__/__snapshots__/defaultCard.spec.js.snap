// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultCard.vue matches snapshot 1`] = `<svgicon-stub name="Aâ™£"></svgicon-stub>`;
